datafusion/src/scalar.rs

```diff
@@ -833,6 +838,74 @@ impl ScalarValue {
         Ok(array)
     }
 
+    fn iter_to_array_list(
+        scalars: impl IntoIterator<Item = ScalarValue>,
+        data_type: &DataType,
+    ) -> Result<GenericListArray<i32>> {
+        let mut offsets = Int32Array::builder(0);
+        if let Err(err) = offsets.append_value(0) {
+            return Err(DataFusionError::ArrowError(err));
+        }
+
+        let mut elements: Vec<ArrayRef> = Vec::new();
+        let mut valid = BooleanBufferBuilder::new(0);
+        let mut flat_len = 0i32;
+        for scalar in scalars {
+            if let ScalarValue::List(values, _) = scalar {
+                match values {
+                    Some(values) => {
+                        let element_array =
+                            ScalarValue::iter_to_array(values.as_ref().clone())?;
```

Comment:
```suggestion
                        let element_array =
                            ScalarValue::iter_to_array(*value)?;
```

I think you can avoid the clone by unboxing the values (I definitely get confused that a `*` deref of a box is how one unboxes values)

Reply:
Ahh, thanks. Done in 1a94255